MapboxNavigation/TopBannerViewController.swift

```diff
@@ -0,0 +1,433 @@
+                child.alpha = 0.0
+            }
+        }) { _ in
+            completion?()
+            for child in self.secondaryChildren {
+                child.isHidden = true
+            }
+        }
+    }
+    
+    public func preview(step stepOverride: RouteStep? = nil, maneuverStep: RouteStep, distance: CLLocationDistance, steps: [RouteStep], completion: CompletionHandler? = nil) {
+        guard !steps.isEmpty, let step = stepOverride ?? steps.first, let index = steps.index(of: step) else {
+            return // do nothing if there are no steps provided to us.
+        }
+        //this must happen before the preview steps are set
+        stopPreviewing(showingSecondaryChildren: false)
+        
+        previewSteps = steps
+        currentPreviewStep = (step, index)
+        
+        
+        
+        guard let instructions = step.instructionsDisplayedAlongStep?.last else { return }
+        
+        let instructionsView = StepInstructionsView(frame: instructionsBannerView.frame)
+        instructionsView.heightAnchor.constraint(equalToConstant: instructionsBannerHeight).isActive = true
+        
+        refreshAppearance(view: instructionsView, padding: topPaddingView)
+        instructionsView.delegate = self
+        instructionsView.distance = distance
+        instructionsView.swipeable = true
+        informationStackView.removeArrangedSubview(instructionsBannerView)
+        instructionsBannerView.removeFromSuperview()
+        informationStackView.insertArrangedSubview(instructionsView, at: 0)
+        instructionsView.update(for: instructions)
+        previewInstructionsView = instructionsView
+        
+        hideSecondaryChildren(completion: completion)
+    }
+    
+    public func stopPreviewing(showingSecondaryChildren: Bool = true) {
+        guard let view = previewInstructionsView else {
+            return
+        }
+        
+        previewSteps = nil
+        currentPreviewStep = nil
+        
+        informationStackView.removeArrangedSubview(view)
+        view.removeFromSuperview()
+        addInstructionsBanner()
+        previewInstructionsView = nil
+        
+        if showingSecondaryChildren {
+            showSecondaryChildren()
+        }
+    }
+    
+    private func refreshAppearance(view: UIView, padding: UIView?) {
+        let viewType = type(of: view)
+        
+        let themedBackgroundColor = viewType.appearance().backgroundColor
+        view.backgroundColor = themedBackgroundColor
+        padding?.backgroundColor = themedBackgroundColor
+    }
+    
+    private func addInstructionsBanner() {
+        informationStackView.insertArrangedSubview(instructionsBannerView, at: 0)
+        refreshAppearance(view: instructionsBannerView, padding: topPaddingView)
+        instructionsBannerView.delegate = self
+        instructionsBannerView.swipeable = true
+    }
+}
+
+// MARK: - NavigationComponent Conformance
+extension TopBannerViewController /* NavigationComponent */ {
```

Comment:
Is this comment needed? Can the NavigationComponent conformance be declared here instead of in the main class definition?